### ISSUE TITLE
Cron deadlock

### ISSUE THREAD

**Issue by stonerl** [owncloud/search_lucene#83]
### Actual behaviour

Lately I observed a strange behavior. I run cron.php from my system every 15 mins, but cron.php did not get executed for days. So I deleted the cron.lock file but after 2 or 3 runs the same error appeard.
So I executed cron.php manually and got this error:

```
PHP Notice:  iconv(): Detected an illegal character in input string in /var/www/owncloud/apps/search_lucene/3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/HTML.php on line 111
```
When this happens the cron.php does not get executed any further, therefore the cron.lock won't be removed which results in a cron deadlock.

### Server configuration
**Operating system:**
Ubuntu 14.04.2 LTS

**Web server:**
Apache/2.4.7 (Ubuntu)

**Database:**
mysql Ver 14.14 Distrib 5.5.41, for debian-linux-gnu (x86_64) using readline 6.3

**PHP version:**
PHP 5.5.9-1ubuntu4.9

**ownCloud version:**
8.0.2

**Comment by DeepDiver1975** [comment]
@VicDeo can you please have a look - THX

**Comment by VicDeo** [comment]
Hm... smells like an old PHP bug http://php.net/manual/en/function.iconv.php#108643

**Comment by VicDeo** [comment]
@stonerl is anything written to owncloud.log additionally?

**Comment by stonerl** [comment]
```
{"reqId":"6077c427d5f8c0c1ddbaf3bc8437401d","remoteAddr":"46.5.18.45","app":"hook","message":"error while running hook (OCA\\Search_Lucene\\Hooks\\Files::indexFile): An exception occurred while executing 'INSERT INTO `oc_lucene_status`(`status`) VALUES(?)' with params [\"N\"]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY'","level":3,"time":"2015-04-20T18:09:18+02:00"}
```

I don't know if this error is anyhow related but I got tons of those entries.

**Comment by VicDeo** [comment]
@stonerl 
Yes, this is definitely related. What version of search lucene do you have?

**Comment by stonerl** [comment]
The one that came with ownCloud 8.0.2. 0.6.0 I suppose.

**Comment by VicDeo** [comment]
@stonerl it's not shipped with OC8

**Comment by stonerl** [comment]
Hmm. I use the ppa for Ubuntu. But anyway, this Version is installed:
![screen shot 2015-04-27 at 23 55 01](https://cloud.githubusercontent.com/assets/2091312/7358866/09232e20-ed39-11e4-9cde-aad8912925da.png)

**Comment by skorvek** [comment]
I'm a bit confused. I am getting the exact same error message.

My cron hasn't been able to run since March. I have had nothing but problems with the search app anyway-- always stalling in my browser and never completing. So I disabled it and tried to run cron from the command line, and I STILL get the same error message!

How is that possible?  It is listed as disabled in my apps.  It is at the same version, installed via the Ubuntu package manager from the suse sources (deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_14.04/ /).

Only after fully "uninstalling" the app, does my cron complete.

Furthermore, it's listed as Recommended after I hit the uninstall button as if it's an internal app.  

Lastly, the official admin documentation references this app under "enabling full text search."

https://doc.owncloud.org/server/8.0/admin_manual/configuration_server/search_configuration.html

Is this an Owncloud app or a 3rd party app?

**Comment by karamatos** [comment]
I have exactly the same problem with ownCloud 8.0.3 on Debian 7.

**Comment by piotr58** [comment]
I think i have the same thing with OC 8.0.3 on Debian 7 also. I have millions lines (more than 300 GB) in owncloud.log like:
`{"reqId":"c361f350b23a8e8bfc88ade2923b75a3","remoteAddr":"","app":"PHP","message":"Uninitialized string offset: 1345613 at \/var\/www\/owncloud_prod\/apps\/search_lucene\/3rdparty\/zendframework\/zendsearch\/library\/ZendSearch\/Lucene\/Storage\/File\/AbstractFile.php#290","level":3,"time":"2015-05-30T21:32:03+00:00 »}`

Is simply disable search_lucene is a good way to solve that until bug solve?

**Comment by VicDeo** [comment]
@piotr58 a bug can't solve itself and I have no idea how to reproduce it.
Your logs looks like https://github.com/owncloud/search_lucene/issues/62 to me.

Could you check if the issue persist with the app version from stable8 branch?

**Comment by stonerl** [comment]
@VicDeo I don't know if this helps but on my server there where files with names like: 漢字undRomaji.txt
A mixture of Japanese Kanjis and western Characters.

**Comment by stonerl** [comment]
I should have explained this a little more. When you use Japanese there are different Characters you can choose from:

1. Kanjis 漢字
2. Hiragana ひらがな
3. Katakana カタカナ
4. Romaji western letters

Katakana can be displayed in half-width: ｶﾀｶﾅ and full-width カタカナ
The same goes for for western Characters ＦＵＬＬ　NORMAL
And there are many more. And you can mix them all together in a file-name. Maybe when such an input gets converted the problem appears.
